api/graphql/fetcher.js

```diff
@@ -40,57 +44,48 @@ export default class Fetcher {
     return this.loaders.queries.load(query).then(instance => {
       if (!instance) return
       this.loaders[tableName].prime(instance.id, instance)
-      return this.format(tableName, instance)
+      return instance
     })
   }
 
-  // once we have an instance, we format it; that means we look at the model
-  // definition and prepare a result object with attributes that match that
-  // definition. we set basic attributes directly, because they have already
-  // been retrieved at this point; but we set up relations as functions, so they
-  // will not run any additional database queries unless the current GraphQL
-  // query specifically asks for them.
-  format (name, instance) {
-    const { model, attributes, getters, relations } = this.models[name]
-    const tableName = model.collection().tableName()
-    if (instance.tableName !== tableName) {
-      throw new Error(`table names don't match: "${instance.tableName}", "${tableName}"`)
-    }
+  _createResolverForModel (name) {
+    const { attributes, getters, relations, model } = this.models[name]
 
-    const formatted = Object.assign(
+    return Object.assign(
       transform(attributes, (result, attr) => {
-        result[camelCase(attr)] = instance.get(attr)
+        result[camelCase(attr)] = x => x[attr] || x.get(attr)
       }, {}),
 
       transform(getters, (result, fn, attr) => {
-        result[attr] = args => fn(instance, args)
+        result[attr] = fn
       }, {}),
 
       transform(relations, (result, attr) => {
         const [ graphqlName, bookshelfName ] = typeof attr === 'string'
           ? [attr, attr] : toPairs(attr)[0]
 
-        const emitter = new EventEmitter()
+        const emitterName = `__${graphqlName}__total_emitter`
 
-        result[graphqlName] = ({ first, cursor, order }) => {
+        result[graphqlName] = (instance, { first, cursor, order }) => {
+          instance[emitterName] = new EventEmitter()
           const relation = instance[bookshelfName]()
           return this.fetchRelation(relation, {first, cursor, order}, instances => {
             const total = instances.length > 0
               ? instances.first().get(PAGINATION_TOTAL_COLUMN_NAME)
               : null
-            emitter.emit('hasTotal', total)
+            instance[emitterName].emit('hasTotal', total)
           })
         }
 
-        result[graphqlName + 'Total'] = () =>
-          new Promise((resolve, reject) => {
-            emitter.on('hasTotal', resolve)
-            setTimeout(() => reject(new Error('timeout')), 6000)
-          })
+        if (model.forge()[bookshelfName]().relatedData.type !== 'belongsTo') {
```

Comment:
this is a fix for a bug that was already present -- we were generating `xTotal` methods for belongsTo relations. this was caught in the switch to `graphql-tools` because it threw an error due to the `communityTotal` (note: singular) field not being defined for the `Membership` type.